### PR TITLE
clowderizing receptor

### PR DESCRIPTION
**Pull request by gburges**

### CODE REVIEW

**Comment by gburges** [comment]
@dehort Hi, can you pls review?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+COPY go.mod go.sum ./
+
+RUN go mod download
+
+COPY . .
+
+USER root
+
+RUN go build -o ws-gateway ./cmd/gateway/main.go
```
Comment (dehort) [inline Dockerfile:13]: Is this supposed to build the image that gets deployed for the gateway and switch?  If so, we need to include the gateway, job-receiver, response_consumer, connection_util, connection_cleaner.

File: deploy/clowdapp.yml [hunk 2]
```diff
@@ -0,0 +1,194 @@
+---
```
Comment (dehort) [inline deploy/clowdapp.yml:1]: We have a cronjob that runs as part of the gateway.  Should the cronjob be listed in this yml file?

I do not see where/how the switch (it runs the job-receiver process) gets started?

File: internal/config/config.go [hunk 3]
```diff
@@ -195,6 +198,14 @@ func GetConfig() *Config {
 		GatewayClusterServiceName:                    options.GetString(GATEWAY_CLUSTER_SERVICE_NAME),
 		PrometheusPushGateway:                        options.GetString(PROMETHEUS_PUSH_GATEWAY),
 	}
+
+	if os.Getenv("CLOWDER_ENABLED") == "true" {
```
Comment (dehort) [inline internal/config/config.go:202]: Please consider hiding this logic behind a method:
https://github.com/RedHatInsights/app-common-go/issues/10

File: cmd/gateway/main.go [hunk 4]
```diff
@@ -78,8 +80,13 @@ func configureConnectionRegistrar(cfg *config.Config, localCM c.ConnectionRegist
 }
 
 func main() {
-	var wsAddr = flag.String("wsAddr", ":8080", "Hostname:port of the websocket server")
-	var mgmtAddr = flag.String("mgmtAddr", ":9090", "Hostname:port of the management server")
+	var wsAddr = *flag.String("wsAddr", ":8080", "Hostname:port of the websocket server")
+	var mgmtAddr = *flag.String("mgmtAddr", ":9090", "Hostname:port of the management server")
+
+	if os.Getenv("CLOWDER_ENABLED") == "true" {
```
Comment (dehort) [inline cmd/gateway/main.go:86]: Please consider hiding this logic behind a method:
https://github.com/RedHatInsights/app-common-go/issues/10

File: internal/config/config.go [hunk 5]
```diff
@@ -195,6 +198,14 @@ func GetConfig() *Config {
...
+		cfg := clowder.LoadedConfig
+		config.RedisHost = cfg.InMemoryDb.Hostname
+		config.RedisPort = strconv.Itoa(cfg.InMemoryDb.Port)
+		config.KafkaBrokers = []string{fmt.Sprintf("%s:%v", cfg.Kafka.Brokers[0].Hostname, *cfg.Kafka.Brokers[0].Port)}
```
Comment (dehort) [inline internal/config/config.go:206]: Please consider hiding this logic behind a method:
https://github.com/RedHatInsights/app-common-go/issues/11

**Review by dehort — COMMENTED** [review]
I left a couple of comments.  I think there might be a few things missing (cronjob, job-reciever, etc).  Let me know what you think.

File: deploy/clowdapp.yml (hunk 2 above)
Comment (gburges) [reply deploy/clowdapp.yml:1]: The switch podSpec starts on line 69, and i added the cronjob

File: internal/config/config.go (hunk 5 above)
Comment (gburges) [reply internal/config/config.go]: added the KafkaServers method

**Comment by dehort** [comment]
@gburges Is it possible to move the clowder related scripts/files into a clowder directory?  I'd like to kind of organize the clowder related files a bit if possible.

**Comment by gburges** [comment]
> @gburges Is it possible to move the clowder related scripts/files into a clowder directory? I'd like to kind of organize the clowder related files a bit if possible.

The clowdapp file is in the deploy directory, the Dockerfile and pr_check afaik needs to live in the app root directory

**Comment by gburges** [comment]
@dehort Hi, can you please take a look at it now?

**Review by dehort — APPROVED** [review]
Looks good to me.

**Pull request merged**